plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -566,9 +566,13 @@ public List<SchemaTableName> listTables(ConnectorSession session, Optional<Strin
     public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
         DeltaLakeTableHandle table = checkValidTableHandle(tableHandle);
-        // This method does not calculate column metadata for the projected columns
-        checkArgument(table.getProjectedColumns().isEmpty(), "Unexpected projected columns");
-        return getColumns(table.getMetadataEntry()).stream()
+        return table.getProjectedColumns()
+                .map(projectedColumns -> (List<DeltaLakeColumnHandle>) projectedColumns.stream()
+                        .map(DeltaLakeColumnHandle.class::cast) // TODO DeltaLakeTableHandle.projectedColumns should be a collection of DeltaLakeColumnHandle
```

Comment:
I will try to remember to solve this in https://github.com/trinodb/trino/pull/17365